apm-collector/apm-collector-worker/src/main/java/org/skywalking/apm/collector/worker/instance/IdentificationCache.java

```diff
@@ -0,0 +1,57 @@
+package org.skywalking.apm.collector.worker.instance;
+
+class IdentificationCache {
+    private static final IdentificationCache INSTANCE = new IdentificationCache();
+
+    State state;
+    IdentificationSegment segment;
+
+    private IdentificationCache() {
+        state = State.NORMAL;
+    }
+
+    public synchronized long fetchInstanceId() {
+        if (state == State.ABNORMAL) {
+            return -1;
+        }
+
+        if (!segment.hasNext()) {
+            IdentificationSegmentFetcher.INSTANCE.fetchSegment(new IdentificationSegmentFetcher.Listener() {
+                @Override
+                public void failed() {
+                    state = State.ABNORMAL;
+                    IdentificationSegmentFetcher.INSTANCE.fetchSegmentInBackGround(INSTANCE);
+                }
+
+                @Override
+                public void success(IdentificationSegment idSegment) {
+                    segment = idSegment;
+                }
+            });
+        }
+
+        return segment.nextInstanceId();
+    }
+
+    public static IdentificationCache initCache() {
+        IdentificationSegmentFetcher.INSTANCE.fetchSegment(new IdentificationSegmentFetcher.Listener() {
+            @Override
+            public void failed() {
+                INSTANCE.state = State.ABNORMAL;
+                IdentificationSegmentFetcher.INSTANCE.fetchSegmentInBackGround(INSTANCE);
```

Comment:
I think you are facing the concurrency situation here. IMO, your `fetchSegmentInBackGround` is not good enough.